bn-apps/cordapp-updates-distribution/corda-updates-app/src/main/kotlin/net/corda/cordaupdates/app/bno/ReportCordappVersionFlowResponder.kt

```diff
@@ -0,0 +1,59 @@
+package net.corda.cordaupdates.app.bno
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.app.member.CordappVersionInfo
+import net.corda.cordaupdates.app.member.ReportCordappVersionFlow
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.ProgressTracker
+import net.corda.core.utilities.unwrap
+
+@InitiatedBy(ReportCordappVersionFlow::class)
+class ReportCordappVersionFlowResponder(private val session : FlowSession) : FlowLogic<Unit>() {
+    @Suspendable
+    override fun call() {
+        val cordappVersionInfo = session.receive<CordappVersionInfo>().unwrap { it }
```

Comment:
Are there really no security checks needed here? Remember that the `.unwrap { it }` part is about forcing you to think about security. I'd really appreciate seeing a comment inside the unwrap block stating why it's OK for this request to be unchecked... for instance, what happens if I crawl the network just invoking this flow on every node?

Reply:
Agree. I've changed this flow to use session filters as well. Stuff like throttling, business network level verification and etc. can go there. I added TODO to the session filter checks to implement them as flow overrides once the feature is available.